numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1851,11 +1870,13 @@ io_unpack(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwds)
 {
     PyObject *obj;
     int axis = NPY_MAXDIMS;
-    static char *kwlist[] = {"in", "axis", NULL};
+    Py_ssize_t count = 0;
+    static char *kwlist[] = {"in", "axis", "count", NULL};
 
-    if (!PyArg_ParseTupleAndKeywords( args, kwds, "O|O&:unpack" , kwlist,
-                &obj, PyArray_AxisConverter, &axis)) {
+    if (!PyArg_ParseTupleAndKeywords(args, kwds, "O|O&O&:unpack" , kwlist,
+                                     &obj, PyArray_AxisConverter, &axis,
+                                     PyArray_SizeConverter, &count)) {
```

Comment:
There has got to be a better way than writing my own 2-bit converter, but oh well...

Reply:
I think PyArray_SizeConverter might be useful, but it probably belongs in a separate MAINT PR, since it has applications elsewhere. I think there's already a function you can call on a `PyObject *` to convert to an `npy_intp`, which would be better to just use in the scope of this PR.

Reply:
Thanks. Could you point me to that function? I wasn't able to find something that matched. `PyArray_intpConverter` seems to do something entirely different.

Reply:
Two reasons that won't work:
1) The signature is wrong: `npy_intp PyArray_PyIntAsIntp(PyObject *o)` instead of `int PyArray_PyIntAsIntp(PyObject *o, npy_intp *x)` required for an argument converter.
2) It specifically forbids `None`, while I specifically want to allow it.

Reply:
Why is it out of scope? I would end up writing the same bit of code anyway, and stashing it where it wouldn't be accessible...

Reply:
If `PyArray_SizeConverter`, it's useful with or without this patch. Conversely, this patch is implementable with or without that converter. There's no reason to bundle the changes together into one review.

Reply:
My main problem with `PyArray_PyIntAsIntp` is that it does not allow `None`. What if I defined the converter locally, without adding it to any public APIs, and later submitted a PR for it if someone else wanted it?

Reply:
Making it a public API would certainly make more sense in a separate review.

Also, you could just use:
```
npy_intp count = 0;
if (count_obj != Py_None) {
    count = PyArray_PyIntAsIntp(count_obj);
    if (error_converting(count)) {
        goto fail;
    }
}
```

I think I like the idea of a similar converter, but I'd like to see it:
* In a separate PR
* With examples elsewhere in numpy showing it is more widely useful

Reply:
Gotcha. Sorry for the slowness.